### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading 'load')

### DISCUSSION THREAD

**Discussion by sundasjay** [gatsbyjs/gatsby#39063] [Help]
Gatsby develop started to suddenly fail on this. 

I have tried updating gatsby-source-wordpress but doesnt help. This only happens when it comes across posts or pages with media in it. If no media, it works fine. 

success  gatsby-source-wordpress  ingest WPGraphQL schema - 1.940s
success  gatsby-source-wordpress  ensuring plugin requirements are met - 0.244s
success createSchemaCustomization - 3.094s
success  gatsby-source-wordpress  Comment - 0.765s - fetched 0
success  gatsby-source-wordpress  Menu - 0.992s - fetched 6
success  gatsby-source-wordpress  MenuItem - 1.012s - fetched 31
success  gatsby-source-wordpress  Category - 1.560s - fetched 7
success  gatsby-source-wordpress  InsuranceSchool - 1.793s - fetched 43
success  gatsby-source-wordpress  PostFormat - 1.997s - fetched 2
success  gatsby-source-wordpress  PressRelease - 2.547s - fetched 15
success  gatsby-source-wordpress  InsuranceSchoolCategory - 2.713s - fetched 6
success  gatsby-source-wordpress  Taxonomy - 2.768s - fetched 5
success  gatsby-source-wordpress  Tag - 3.190s - fetched 11
success  gatsby-source-wordpress  Post - 3.317s - fetched 43
success  gatsby-source-wordpress  MediaItems - 2.812s - fetched 0
success  gatsby-source-wordpress  fetch root fields - 3.380s
success  gatsby-source-wordpress  Page - 4.177s - fetched 34
success  gatsby-source-wordpress  TestimonialCategory - 4.418s - fetched 5
success  gatsby-source-wordpress  ContentType - 5.505s - fetched 6
success  gatsby-source-wordpress  UserRole - 0.188s - fetched 0
success  gatsby-source-wordpress  User - 0.380s - fetched 2
success  gatsby-source-wordpress  Testimonial - 2.629s - fetched 73

 ERROR  UNKNOWN

Cannot read properties of undefined (reading 'load')

  TypeError: Cannot read properties of undefined (reading 'load')

  - process-node.js:292 map
    [gatsby-starter-wordpress-homepage]/[gatsby-source-wordpress]/src/steps/source-nodes/create-nodes/process-node.js:292
    :23

  - Array.map

  - process-node.js:324 getCheerioElementsFromMatches
    [gatsby-starter-wordpress-homepage]/[gatsby-source-wordpress]/src/steps/source-nodes/create-nodes/process-node.js:324
    :6

  - process-node.js:475 nodeStringFilter
    [gatsby-starter-wordpress-homepage]/[gatsby-source-wordpress]/src/steps/source-nodes/create-nodes/process-node.js:475
    :27

  - process-node.js:943 processNodeString
    [gatsby-starter-wordpress-homepage]/[gatsby-source-wordpress]/src/steps/source-nodes/create-nodes/process-node.js:943
    :24

  - process-node.js:981 processNode
    [gatsby-starter-wordpress-homepage]/[gatsby-source-wordpress]/src/steps/source-nodes/create-nodes/process-node.js:981
    :31

  - create-nodes.js:45
    [gatsby-starter-wordpress-homepage]/[gatsby-source-wordpress]/src/steps/source-nodes/create-nodes/create-nodes.js:45:
    33

  - index.js:163 run
    [gatsby-starter-wordpress-homepage]/[p-queue]/dist/index.js:163:29

not finished source and transform nodes - 8.272s
not finished  gatsby-source-wordpress  fetching nodes - 8.233s - 289 total
not finished Downloading remote files - 4.982s
not finished  gatsby-source-wordpress  creating nodes - 0.098s
not finished  gatsby-source-wordpress  MediaItem - 0.107s

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by DTr-Nghia** [comment]
I'm facing this issue too, any update on this?

**Comment by justinaht** [comment]
Hi all,

We have a site but now also cannot be built
The same problem occurs

Diagnosis: The build failure is caused by the following error in the logs:
Cannot read properties of undefined (reading 'load')

```
error UNHANDLED REJECTION Cannot read properties of undefined (reading 'load')


  TypeError: Cannot read properties of undefined (reading 'load')
  
  - process-node.js:290 map
    [repo]/[gatsby-source-wordpress]/src/steps/source-nodes/create-nodes/process    -node.js:290:23
  
  - Array.map
  
  - process-node.js:322 getCheerioElementsFromMatches
    [repo]/[gatsby-source-wordpress]/src/steps/source-nodes/create-nodes/process    -node.js:322:6
  
  - process-node.js:473 nodeStringFilter
    [repo]/[gatsby-source-wordpress]/src/steps/source-nodes/create-nodes/process    -node.js:473:27
  
  - process-node.js:939 processNodeString
    [repo]/[gatsby-source-wordpress]/src/steps/source-nodes/create-nodes/process    -node.js:939:24
  
  - process-node.js:977 processNode
    [repo]/[gatsby-source-wordpress]/src/steps/source-nodes/create-nodes/process    -node.js:977:31
  
  - create-nodes.js:45 
    [repo]/[gatsby-source-wordpress]/src/steps/source-nodes/create-nodes/create-    nodes.js:45:33
  
  - index.js:163 run
    [repo]/[p-queue]/dist/index.js:163:29
  

not finished source and transform nodes - 1951.067s
not finished  gatsby-source-wordpress  fetching nodes - 1949.787s - 5794 total
not finished Downloading remote files - 1945.736s
not finished  gatsby-source-wordpress  creating nodes - 0.388s
not finished  gatsby-source-wordpress  MediaItem - 0.387s
Failed during stage 'building site': Build script returned non-zero exit code: 2 (https://ntl.fyi/exit-code-2)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! site@2.0.1 build: `cross-env GATSBY_CONCURRENT_DOWNLOAD=100 gatsby build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the site@2.0.1 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /opt/buildhome/.npm/_logs/2024-08-20T16_51_41_035Z-debug.log
error Command failed with exit code 1. (https://ntl.fyi/exit-code-1)
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Step starting.
Step started.
Step ended.
​
"build.command" failed                                        
────────────────────────────────────────────────────────────────
​
  Error message
  Command failed with exit code 1: yarn install --ignore-engines && yarn build:clean (https://ntl.fyi/exit-code-1)
​
  Error location
  In Build command from Netlify app:
  yarn install --ignore-engines && yarn build:clean
​
  Resolved config
  build:
    base: /opt/build/repo/site
    command: yarn install --ignore-engines && yarn build:clean
    commandOrigin: ui
    environment:
      - CLIENT_EMAIL
      - CLIENT_ID
      - FB_APP_ID
      - GATSBY_CONCURRENT_DOWNLOAD
      - GATSBY_CPU_COUNT
      - GATSBY_DISABLE_CACHE_PERSISTENCE
      - GATSBY_DISQUS_NAME
      - GATSBY_EXPERIMENTAL_QUERY_CONCURRENCY
      - GATSBY_HOT_LOADER
      - GATSBY_JPEG_ENCODER
      - GATSBY_SITE_URL
      - GATSBY_WATCHDOG_STUCK_STATUS_TIMEOUT
      - GOOGLE_TAGMANAGER_ID
      - GTM_CONTAINER_ID
      - NETLIFY_BUILD_DEBUG
      - NODE_ENV
      - NODE_OPTIONS
      - NODE_VERSION
      - NPM_VERSION
      - PASSWORD
      - PRESERVE_FILE_DOWNLOAD_CACHE
      - PRIVATE_KEY
      - PRIVATE_KEY_ID
      - PROJECT_ID
      - SPREADSHEET_ID
      - USER_NAME
      - WORDPRESS_SITE_URL
      - YARN_VERSION
    publish: /opt/build/repo/site/public
    publishOrigin: ui
  plugins:
    - inputs: {}
      origin: ui
      package: '@netlify/plugin-gatsby'
Build failed due to a user error: Build script returned non-zero exit code: 2
Failing build: Failed to build site
```

We have tried to upgrade the plugins but the issue is still there

Kindly advise,
Thanks!

**Comment by raaiqwilliams** [comment]
I have also been experiencing this across multiple websites over the past few weeks, we had initially thought it was an issue with the gatsby sharp plugin but even updating this and all other dependencies still leads to the error :(

**Comment by WarboxLiam** [comment]
Anyone have an update for this?

**Comment by pratiweb** [comment]
@WarboxLiam 

After updating gats by latest module This error is resolved.

**Reply by ankurparihar** [reply]
+1, for some reason `cheerio` package import was coming as undefined, reinstalling node_modules fixed it for me.
